homeassistant/helpers/sun.py

```diff
@@ -99,6 +99,59 @@ def get_location_astral_event_next(
         mod += 1
 
 
+@callback
+@bind_hass
+def get_astral_event_previous(
```

Comment:
Would this be better as functions: `get_astral_event` and `get_location_astral_event` with a `next` boolean argument to indicate next or previous? A lot of the code between the two functions is common. The existing functions `get_astral_event_next` and `get_location_astral_event_next` could just call the more general function with `next=True`.

Reply:
Thanks for your quick review. I considered that but was worried that it'd make the logic harder to read and understand. For example, it'd lead to lines similar to:

```py
return dt < utc_now if next else dt > utc_now
```

I'm ok refactoring as such if you still believe it's the right choice for the codebase. What do you think?

Reply:
@Swamp-Ig Sorry didn't mention you in my prior reply. What are your thoughts on path forward?

Reply:
I don't mind. Although I am listed as a code owner, I don't actually do a lot of HA work anymore.

Unless one of the other devs has a strong opinion it LGTM.